uaclient/cli.py

```diff
@@ -248,21 +255,108 @@ def status_parser(parser):
+
+        * name: The name of the entitlement
+        * cfg: A UAClient config object
+
+      The additional parameters are handled by the extra_params dict.
+
+    :param cfg: An UAClient config object.
+    :param entitlements_found:
+       A list of valid entitlements to apply the action on.
+    :param entitlements_not_found:
+       A list of entitlements that are invalid, meaning that
+       they are not provide by the tool.
+    :param extra_params:
+       A dict containing the additional parameters the action
+       function may require.
+    """
+    entitlements_not_disabled = []
+    entitlements_disabled = []
+
+    ret = len(entitlements_not_found) == 0
+
+    for entitlement_name in entitlements_found:
+        print(entitlement_name)
+
+        if action_fn(entitlement_name, cfg, **extra_params):
+            ret &= True
+            entitlements_disabled.append(entitlement_name)
+        else:
+            ret = False
+            entitlements_not_disabled.append(entitlement_name)
+
+        print()
+
+    print(
+        ua_status.action_report(
+            action_name=action_name,
+            entitlements_not_found=entitlements_not_found,
+            entitlements_not_succeeded=entitlements_not_disabled,
+            entitlements_succeeded=entitlements_disabled,
+        ),
+        end="",
+    )
+
+    return 0 if ret else 1
+
+
+def _perform_disable(entitlement_name, cfg, *, assume_yes):
     """Perform the disable action on a named entitlement.
 
-    @return: 0 on success, 1 otherwise
+    :param entitlement_name: the name of the entitlement to enable
+    :param cfg: the UAConfig to pass to the entitlement
+    :param assume_yes:
+        Assume a yes response for any prompts during service enable
+
+    @return: True on success, False otherwise
     """
-    ent_cls = entitlements.ENTITLEMENT_CLASS_BY_NAME[args.name]
-    entitlement = ent_cls(cfg, assume_yes=args.assume_yes)
-    ret = 0 if entitlement.disable() else 1
+    ent_cls = entitlements.ENTITLEMENT_CLASS_BY_NAME[entitlement_name]
+    entitlement = ent_cls(cfg, assume_yes=assume_yes)
+    ret = entitlement.disable()
     cfg.status()  # Update the status cache
     return ret
 
 
+@assert_root
+@require_valid_entitlement_names("disable")
+@assert_attached(ua_status.MESSAGE_ENABLE_FAILURE_UNATTACHED_TMPL)
+def action_disable(args, cfg, **kwargs):
+    """Perform the disable action on a list of entitlements.
+
+    @return: 0 on success, 1 otherwise
+    """
+    entitlements_not_found = kwargs.get("entitlements_not_found", [])
+    entitlements_found = kwargs.get("entitlements_found", [])
+    extra_params = {}
+    extra_params["assume_yes"] = args.assume_yes
+
+    return perform_actions_on_service_entitlements(
+        action_name="disabled",
+        action_fn=_perform_disable,
+        cfg=cfg,
+        entitlements_found=entitlements_found,
+        entitlements_not_found=entitlements_not_found,
+        **extra_params
+    )
```

Comment:
Passing the entitlements_found and entitlements_not_found down into a new function in order to run a for loop feels like a sign that we're pushing complexity into a new function that may not need to be there.

What do you think about just keeping the looping constructs here in action_disable?
we also could drop the entitlements_not_found variable from both the decorator and and this function as we can derive that locally with  **entitlements_not_found = list(set(args.names) - set(entitlements_found))**

```suggestion
    entitlements_not_found = set(args.names) - set(entitlements_found))
    for entitlement in entitlements_found:
        _perform_disable(entitlement_name, cfg, assume_yes=args.assume_yet)
    if entitlements_not_found:
       raise exceptions.UserFacingError(
￼             tmpl.format(operation="disabled, name=", ".join(entitlements_not_found))
￼       )
```

Reply:
That makes sense, I will update the code